headlamp-k8s/headlamp
# backend: refactor listing all contexts and error

This removes the dependacy of manually parsing of k8s resources. It was a problem as we had to be in sync with k8s API changes, which is not a good way. This change removes that and it splits kubeconfig into seperate contexts and validates them.

### Testing

- In your kubeconfig add some context with partial validity. For reference please check [here](https://github.com/headlamp-k8s/headlamp/blob/e13376a49d4191032926373e11d30a1a6c5ff4f7/backend/pkg/kubeconfig/test_data/kubeconfig_partialcontextvalid)
- Run backend `cd backend && go run ./cmd -dev`
- Run frontend `cd frontend && npm start`
- You should see errors in backend and all clusters in frontend.
- You can try with different kubeconfig and check.

### Reference

https://github.com/headlamp-k8s/headlamp/pull/2289#issuecomment-2343545934
